### PR TITLE
Added NIST MT evaluation metric and added NIST international_tokenize() 

### PR DESCRIPTION
**Pull request by alvations**
Added the NIST score, based on http://dl.acm.org/citation.cfm?id=1289273 and [mteval-14.pl](https://github.com/moses-smt/mosesdecoder/blob/master/scripts/generic/mteval-v14.pl#L889) implementation

I'm unsure of whether the final arithmetic mean computation is correct, i.e. https://github.com/alvations/nltk/blob/develop/nltk/translate/nist_score.py#L121 . 

~Should it be `return bp * sum(p_n)` or `return bp * sum(map(math.log, p_n))`? If anyone knows the right answer, please help to clarify. Thanks in advance!~

### CODE REVIEW

**Comment by alvations** [comment]
I've squashed the commits and corrected the arithmetic mean implementation. 

This PR is ready for review =)

**Comment by alvations** [comment]
[CI: retest]

File: nltk/tokenize/nist.py [hunk 1]
```diff
@@ -34,9 +38,32 @@ class NISTTokenizer(TokenizerI):
...
+    >>> nist.tokenize(s, lowercase=True) == expected_lower  # Lowercased.
+    True
+
+    The international_tokenize() is the preferred function when tokenizing
+    non-european text, e.g.
+
+    >>> from nltk.tokenize.nist import NISTTokenizer
+    >>> nist = NISTTokenizer()
+
+    # Intput strings.
```
Comment (stevenbird) [inline nltk/tokenize/nist.py]: Input strings

File: nltk/tokenize/nist.py [hunk 2]
```diff
@@ -87,8 +143,25 @@ def tokenize(self, text, preserve_case=True,
...
+    def international_tokenize(self, text, lowercase=False,
+                               split_non_ascii=True,
+                               return_str=False):
         text = text_type(text)
-        # Language independent regex.
-        text = self.lang_independent_sub(text)
-        #
+        # Different from the 'normal' tokenize(), STRIP_EOL_HYPHEN is applied
+        # first before unescaping.
+        regexp, subsitution = self.STRIP_SKIP
```
Comment (stevenbird) [inline nltk/tokenize/nist.py]: substitution

File: nltk/tokenize/nist.py [hunk 3]
```diff
@@ -55,6 +82,35 @@ class NISTTokenizer(TokenizerI):
     LANG_DEPENDENT_REGEXES = [PUNCT, PERIOD_COMMA_PRECEED,
                               PERIOD_COMMA_FOLLOW, DASH_PRECEED_DIGIT]
 
+    # Perluniprops characters used in NIST tokenizer.
+    Number = text_type(''.join(set(perluniprops.chars('Number')))) # i.e. \p{N}
```
Comment (stevenbird) [inline nltk/tokenize/nist.py]: Unusual to have uppercase variable name?

**Comment by stevenbird** [comment]
@alvations: you could identify yourself in the code as author (then after your name, say porting from Perl or whatever; also name the Perl authors?)

File: nltk/tokenize/nist.py (hunk 3 above)
Comment (alvations) [reply nltk/tokenize/nist.py]: Using lowercase, those were bad habits ported from the perl script ;P

**Comment by alvations** [comment]
Actually, I've no idea who wrote the NIST tokenizer, I would assume it's Doddington himself years ago but it's unclear. 

But it's so tied up with original BLEU from Papineni 16 years ago that they cite him as the original author of the nist mteval script.

See ftp://jaguar.ncsl.nist.gov/mt/resources/mteval-v14.pl

**Pull request merged**